### ISSUE TITLE
Issue with React hook form

### ISSUE THREAD

**Issue by gtwebsite** [sanity-io/sanity-plugin-media#58]
My app already have React Hook Form version 7x, and seems this has a conflict with it.

```
Warning in /nx/client/node_modules/@hookform/resolvers/yup/dist/yup.module.js
1:1131-1132 "export 'transformToNestObject' (imported as 'e') was not found in 'react-hook-form'
 @ /nx/client/node_modules/@hookform/resolvers/yup/dist/yup.module.js
```

Greatly appreciated if it got a fix.

**Comment by ElektrikSpark** [comment]
Am running into this with Sanity Studio v3. I am using `"react-hook-form": "^6.15.1",` until this gets fixed.

**Comment by gtwebsite** [comment]
To fix, override the version in package.json when using npm 8

**Comment by ElektrikSpark** [comment]
Awesome! Thanks! 🎉

**Comment by ElektrikSpark** [comment]
Actually, it stopped working for me. I pushed to Github and the build failed with the same error. Can you send an example of your `package.json` with the `overrides`?

I have tried:
`"overrides": {
    "react-hook-form": "^7.39.7"
  }`

`"overrides": {
    "sanity-plugin-media": {
      "react-hook-form": "^6.15.1"
    }
  }`

`"overrides": {
    "react-hook-form@<7": "6.15.1",
    "react-hook-form@>7": "^7.39.7"
  }`

Thanks!

**Comment by ElektrikSpark** [comment]
On the npm overrides [docs](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#overrides).

`You may not set an override for a package that you directly depend on unless both the dependency and the override itself share the exact same spec. To make this limitation easier to deal with, overrides may also be defined as a reference to a spec for a direct dependency by prefixing the name of the package you wish the version to match with a $.`

Looks like the only thing I can do is:
`"overrides": {
    "react-hook-form": "$react-hook-form"
  }`

But it doesn't work 🤷‍♂️

**Comment by ElektrikSpark** [comment]
Settled on using an alias.

`"react-hook-form-latest": "npm:react-hook-form@^7.39.7"`

**Comment by ajereos-circadence** [comment]
this depends on the npm version (8.5.5), mine looks like this

```
"overrides": {
    "sanity-plugin-media": {
      "react-hook-form": "~7.20.2",
      "@hookform/resolvers": "2.8.5"
    }
}
```

**Comment by joshwcorbett** [comment]
tried overriding but to no avail, would love to see this package get upgraded so workarounds like this aren't necessary

**Comment by robinpyon** [comment]
Thanks all – really appreciate your patience here!

This should be fixed in `2.1.1`, which now uses an updated version of react-hook-form and no longer yields peer dependency warnings on install.
